File: Pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // take money input and prompt user again if negative
    int money;
    do
    {
        money = get_int("Change owed: ");
    }
    while (money < 0);
    // different coins used

    int quarters;
    int dimes;
    int nickels;
    int cents;

    // calculation of number of each type of coins
    quarters = money / 25;
    money = money - (25 * quarters);

    dimes = money / 10;
    money = money - (10 * dimes);

    nickels = money / 5;
    money = money - (5 * nickels);

    cents = money;

    int coins;
    coins = cents + nickels + dimes + quarters;

    printf("%i\n", coins);
}

File: Pset6/sentimental-cash/cash.py
import cs50
import math

# only accept if money is greater than 0
while True:
    money = cs50.get_float("Enter change: ")
    if money > 0:
        break
# Ensure dollars converted to cents + initiate all variables
money = money * 100
quarters = 0
dimes = 0
nickels = 0
cents = 0
coins = 0
# Calculate each coin's value and subtract from our total number of cents
quarters = math.floor(money / 25)
money = money - (25 * quarters)
dimes = math.floor(money / 10)
money = money - (10 * dimes)
nickels = math.floor(money / 5)
money = money - (5 * nickels)
cents = money
# Add all of the coins in the previous part
coins = quarters + dimes + nickels + cents
# print out the number of coins
print(coins)
